tests/endpoint_decorators_test.py

```diff
@@ -42,3 +42,38 @@ def test_log_headers(*args, **kwargs):
                 "Authorization: REDACTED\n"
                 "authorization: REDACTED\n"
                 "SafeStuff: This better be logged this way")
+
+    def test_logbodies_errors_on_invalid_json(self):
+        """
+        """
+        data = "{\"invalid\": ...}"
+        valueErrorStr = "No JSON object could be decoded"
+        logger_mock = Mock()
+        logger_mock.info = Mock()
+        logger_mock.exception = Mock()
+        request_mock = type("FakeRequest", (), {
+            "headers": {
+                "content-type": "application/json"
+            },
+            "data": data
+        })
+        request_mock.get_data = Mock(return_value=data)
+        mock_container = type("FakeContainer", (), {
+            "logger": logger_mock,
+            "request": request_mock
+        })
+
+        @decorators.logbodies
+        def test_log_bodies(*args, **kwargs):
+            return request_mock
+
+        test_log_bodies(mock_container)
+        logger_mock.info.assert_any_call("Invalid JSON from request.")
+
+        # Get the call arguments from logger_mock.exception,
+        # and assert if the first argument is a ValueError with
+        # the correct error message.
+        args, kwargs = logger_mock.exception.call_args
+        callArg = args[0]
+        self.assertTrue(callArg.message == valueErrorStr)
+        self.assertTrue(isinstance(callArg, ValueError))
```

Comment:
I don't like the assertions since, I would rather do assertEqual so it gives you info when it fails. Also expected comes first usually.
```
callArg = logger_mock.exception.call_args[0][0]
self.assertEqual(valueError, type(callArg))
self.assertEqual(valueErrorStr, callArg.message)
```